### PR TITLE
use conditional requirements for backport libraries closes #147

### LINKED ISSUE #147: Dependency that is only compatible/needed with/in python 2 in setup.py
this causes pip install to fail (at least with my py3.4 system)

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

File: vcr/compat.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+try:
+    import unittest.mock
```
Comment (IvanMalison) [inline vcr/compat.py]: shouldn't this be 
```
from unittest import mock
```
?

**Comment by IvanMalison** [comment]
installs are failing with
error in vcrpy setup command: Invalid environment marker: python_version<=2.4

**Comment by IvanMalison** [comment]
pkg_resources.DistributionNotFound: contextlib2

**Comment by graingert** [comment]
I've arsed this up pretty bad. But I know how to fix this. Please bare with me 😻

**Comment by IvanMalison** [comment]
Hah no worries. Thanks for the pr. This issue was annoying me.

**Comment by IvanMalison** [comment]
closes #147

**Comment by graingert** [comment]
@IvanMalison woot! they pass now

**Comment by kevin1024** [comment]
Awesome (+1)

**Pull request merged**